scripts/release_candidate.py

```
@@ -694,6 +766,113 @@ def _validate_resolver_assets(directory: Path, version: str) -> None:
+    validate_staged_mode = snapshots is not None
+    if snapshots is None:
+        snapshots = {name: _validated_installer_source(name) for name in expected_names}
+    elif set(snapshots) != set(expected_names):
+        raise CandidateError("reviewed installer snapshots do not match the release asset set")
+    for name in installer_asset_names(version):
+        snapshot = snapshots[name]
+        candidate = directory / name
+        try:
+            candidate_info = candidate.lstat()
+        except OSError:
+            raise CandidateError(f"release candidate is missing installer asset {name}")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)